multus/multus.go

```diff
@@ -202,9 +202,23 @@ func delegateAdd(exec invoke.Exec, ifName string, delegate *types.DelegateNetCon
 
 			cniArgs = fmt.Sprintf("%s;IP=%s", cniArgs, delegate.IPRequest)
 			logging.Debugf("Set IP address %q to %q", delegate.IPRequest, ifName)
+
+			if delegate.ExtraFields != nil {
+				for key, value := range delegate.ExtraFields {
+					if strings.Contains(key, ";") {
+						return nil, logging.Errorf("Key %q contains ';' character which is invalid", key)
+					}
+					if strings.Contains(value, ";") {
+						return nil, logging.Errorf("Value %q contains ';' character which is invalid", value)
+					}
+					cniArgs = fmt.Sprintf("%s;%s=%s", cniArgs, key, value)
```

Comment:
I think you need also to check if the key or value contains `=` this can also break the code.